core/remotes/docker/resolver.go

```diff
@@ -371,7 +371,7 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 					req.header[key] = append(req.header[key], value...)
 				}
 
-				resp, err := req.doWithRetries(ctx, nil)
+				resp, err := req.doWithRetries(ctx, nil, true)
```

Comment:
`true` -> `i == len(hosts)-1`?

Reply:
Same as mentioned above, we don't need to give up quickly since we already got a response from this host.